chess-AI/ai_io.py

```diff
@@ -2,64 +2,134 @@
-    """Load model parameters into given network from external file
+    # make sure a path was specified
+    if ds_saving['save_path'] is None:
+        return
+        
+    date_string = create_date_string()
+    # get the full file path to save
+    full_path = ds_saving['save_path'] + "dataset-" +  date_string + ".pt"
+    # save the dataset
+    torch.save(dataset, full_path)
+
+    if ds_saving['figshare_save']:
+        # Save dataset to figshare
+        title = "dataset-" + date_string
+        desc = "This description is a placeholder"
+        keys = ["Dataset", "Chess"]
+        # 179 is category [Artificial Intelligence and Image Processing]
+        categories = [179]
+        figshare_api.upload(title, desc, keys, categories, full_path)
+
+def save_model(nnet, m_saving, checkpointing, file_name=None):
+    """Save a model to Figshare or Locally.
+
+    The save_path refers to the directory where the file will be saved. For saving to Figshare,
+    the file still needs to be saved locally first.
     """
-    if model_path is not None:
-        nnet.load_state_dict(torch.load(model_path))
-    else:
-        for i in range(num_saved_models):
-            if not os.path.isfile(f'./models-{i + 2}.pt'):
-                if i != 0:
-                    nnet.load_state_dict(torch.load(f'./models-{i + 1}.pt'))
-                break
-
+    # A save path must be specified for the model to be saved
+    if m_saving['save_path'] is not None:
+        # Check if model is being saved as part of checkpointing
+        if checkpointing:
+            assert file_name is not None, " For checkpointing, a file name msut be specified"
+            full_path = m_saving['checkpoint_path'] + file_name + ".pt"
+            torch.save(nnet.state_dict(),full_path)
+        else:
+            date_string = create_date_string()
+            full_path = m_saving['save_path'] + "model-" +  date_string + ".pt"
+            torch.save(nnet.state_dict(), full_path)
+            # Check if model should be saved to Figshare
+            if m_saving['figshare_save']:
+                title = "model-" + date_string
+                desc = "This description is a placeholder"
+                keys = ["Neural Network", "Chess"]
+                # 179 is category, Artificial Intelligence and Image Processing
+                categories = [179]
+                api = FigshareApi()
+                api.upload(title, desc, keys, categories, full_path)
+
+def load_model(nnet, m_saving, show_dash):
+    """Load model parameters from Figshare or locally."""
+    # A load path must be specified to load a file
+    if m_saving['load_path'] is not None:
+        # Check if model should be loaded from Figshare
+        if m_saving['figshare_load']:
+            base_path,file_name = file_from_path(m_saving['load_path'])
+            # Ensure file exists in figshare
+            assert figshare_api.get_figshare_article(m_saving['load_path'], base_path, file_name), \
```

Comment:
this function takes two arguments, but it's being passed 3. I'm not sure which ones are supposed to be passed, but can u pls update either way?

Reply:
Whoops , fixed that